src/net/java/sip/communicator/impl/protocol/jabber/extensions/colibri/ColibriConferenceIQ.java

```diff
@@ -2271,13 +2361,63 @@ public void setDisplayName(String displayName)
         }
 
         /**
+         * Sets the transport element of this <tt>ChannelBundle</tt>.
+         * @param transport the transport to set.
+         */
+        public void setTransport(IceUdpTransportPacketExtension transport)
+        {
+            this.transport = transport;
+        }
+
+        /**
          * Sets the ID of this <tt>Endpoint</tt>.
          * @param id the ID to set.
          */
         public void setId(String id)
         {
             this.id = id;
         }
+
+        /**
+         * Sets the stats ID of this <tt>Endpoint</tt>.
+         * @param statsId the stats ID to set.
+         */
+        public void setStatsId(String statsId)
+        {
+            this.statsId = statsId;
+        }
+
+        /**
+         * Appends the XML <tt>String</tt> representation of this
+         * <tt>Endpoint</tt> to <tt>xml</tt>.
+         *
+         * @param xml the <tt>StringBuilder</tt> to which the XML
+         * <tt>String</tt> representation of this <tt>Endpoint</tt> is to be
+         * appended
+         */
+        public IQChildElementXmlStringBuilder toXML(
+            IQChildElementXmlStringBuilder xml)
+        {
+            xml.halfOpenElement(ELEMENT_NAME)
+                .attribute(ID_ATTR_NAME, id);
+
+            xml.optAttribute(DISPLAYNAME_ATTR_NAME, displayName);
+            xml.optAttribute(STATS_ID_ATTR_NAME, statsId);
+
+            IceUdpTransportPacketExtension transport = getTransport();
```

Comment:
Forget about all that when `addExtension` is called from within `setTransport`, just call `setEmptyElement` here.

NEVER call `closeElement`, Smack does this for you. Have a look at [getChildElementXML()](https://github.com/igniterealtime/Smack/blob/master/smack-core/src/main/java/org/jivesoftware/smack/packet/IQ.java#L156), this might clear things up. It behaves quite differently than just doing plain XML construction.

Reply:
A unit test based how the element should look like wouldn't hurt (similar to e.g. `MuteIqProviderTest` in Jicofo, which should actually be somewhere here in Jitsi)

Reply:
Ok, so you want to refactor all the logic in the ColibriConferenceIQ before doing these changes first?

Reply:
No, why?